gocse/README.md

```diff
@@ -0,0 +1,52 @@
+# Go CSFLE Example
+
+## Steps
+
+1. Clone this repository:
+
+   ```sh
+     git clone https://github.com/mongodb-university/csfle-guides.git
+     cd csfle-guides/gocse
+   ```
+
+   Work from the gocse directory for the remainder of these
+   instructions.
+
+2. Start a `mongod` instance (Enterprise version >= 4.2) running on port 27017
+
+3. Make sure you have the `master-key.txt` file in the root of your
+   execution environment. This is a 96-byte cryptographically-secure generated
+   master encryption key required to run this example project. To generate your
+   own master key or use a KMS, refer to the [CSFLE Use Case
+   Guide](https://docs.mongodb.com/drivers/security/client-side-field-level-encryption-guide/).
+
+   The settings for each supported KMS are located in
+   `kms/provider.go`. In this project we have used [godotenv](https://github.com/joho/godotenv) to
+   load our environmental variables. Feel free to replace the calls to `GetCheckedEnv` with string values if you
+   prefer.
+
+4. If `mongocryptd` is not installed in your system path, ensure you specify its
+   location in `csfle/encrypted_client.go` in the `EncryptedClient` function.
+
+5. Uncomment the initializer for hte KMS provider you would like to use in `main.go`
+
+   ```go
+   // awsKMS := kms.AWSProvider()
+   // azureKMS := kms.AzureProvider()
+   // gcpKMS := kms.GCPProvider()
+   localKMS := kms.LocalProvider(localKey)
+
+   ```
+
+6. Run the `go run -tags=cse .`. This does the following:
+
+   - Reads in the local key from `master-key.txt`
+   - Reads in configuration information for your preferred KMS.
+   - Fetches the master key from the KMS and creates a data key in the **keyVault** collection.
```

Comment:
Do you think the KMS terminology, especially when referring to a local master key, might be confusing for users that haven't read the document on KMS providers?

Reply:
I've spelled out the acronyms. It may be a bit confusing but I don't want to duplicate the information from the guide to the readme too much. We do link back to the guide, and I think it's fair to expect users to refer to centralized documentation. Thoughts?

Reply:
To clarify, I'm referring to the term "KMS".
In the initial guide (https://docs.mongodb.com/drivers/security/client-side-field-level-encryption-guide), we don't use the term much. We also use it incorrectly when we name a variable "kmsProvider" which references the contents of the local key. I wonder if in that case it would be more correct to call it a "key provider" rather than a full key management system (https://en.wikipedia.org/wiki/Key_management#Key_management_system).

Reply:
I've made it clear there are parallel paths and a combined path, and cleaned up the wording a bit for using a local master key.